### PR TITLE
feat: Wasm32 target custom font loading 

### PR DESCRIPTION
**Pull request by antmelnyk**
Heyo!

I know there is an [opened but a bit stale PR](https://github.com/yisibl/resvg-js/pull/118) that improves font handling, but I feel like ResvgJS lacks a direct, essential font-loading support on the wasm32 target before going into some more complicated APIs.

What do you think about the solution presented in this PR? Dead simple, with no API breaking change, just new optional font options and the possibility to pass an array of font data.

Works like this:
```javascript
let resvg_wasm = await fetch("./index_bg.wasm");
await initWasm(resvg_wasm);

// Load font as buffer
const req = new XMLHttpRequest();
req.open("GET", "./Roboto-Regular.ttf", true);
req.responseType = "arraybuffer";

// Make sure it's a Uint8Array
const arrayBuffer = req.response;
const font = new Uint8Array(arrayBuffer);

const opts = {
  font: {
    fontsBuffers: [font],
  },
};

const resvgJS = new Resvg(svg, opts);
```

In the future, more complex API can be built on top of it or just, well, replaced if there is a better solution at one point.

### CODE REVIEW

**Comment by zimond** [comment]
I like this PR and in fact it does not conflict with the original font resolving PR, which focuses on resolving font related attributes in the SVG so that the users could get a hint about which font buffer to load. And this PR simply provides a way to load the buffer.

**Comment by antmelnyk** [comment]
I agree. Those are orthogonal changes!

The only problem I see now is that the WASM file size increased by 600 KBs, probably because of the addition of the `text` feature from Resvg. I wonder if something can be done about it...

**Comment by yisibl** [comment]
Thank you for your contribution, this is a great addition. I expect this wasm file to be generated and distributed independently, given the file size limitations for certain scenarios.

I have been very busy lately and will look at this PR in April.

**Comment by yisibl** [comment]
@antmelnyk Can you add some test cases?

**Comment by shuding** [comment]
Looking forward to seeing this being landed too, as we will likely to rely on this feature to solve https://github.com/vercel/satori/issues/186 👍

**Comment by zimond** [comment]
@antmelnyk can you add some test cases covering this piece of code? Thanks. Lacking tests is what blocking this PR being merged.

**Comment by zimond** [comment]
@antmelnyk lint is failing

File: wasm/index.d.ts [hunk 1]
```diff
@@ -66,24 +67,44 @@ export type ResvgRenderOptions = {
...
+} & FontOptions;
 /**
  * Initialize Wasm module
  * @param module_or_path WebAssembly Module or .wasm url
  *
  */
 export declare const initWasm: (module_or_path: Promise<InitInput> | InitInput) => Promise<void>;
 export declare const Resvg: {
-	new (svg: Uint8Array | string, options?: ResvgRenderOptions | undefined): {
+	new (svg: Uint8Array | string, options?: ResvgRenderOptions): {
```
Comment (yisibl) [inline wasm/index.d.ts:96]: Keep `| undefined`

**Comment by yisibl** [comment]
> Please rerun all the necessary builds for release with the latest dependencies because I don't know precisely what must be run. There are a lot of build and bundle commands in package.json.

You just need to run two sets of commands:

```
 npm run build && npm run test
 npm run build:wasm && npm run test:wasm
```

File: index.d.ts [hunk 2]
```diff
@@ -29,27 +40,6 @@ export type ResvgRenderOptions = {
...
-export type CustomFontsOptions = {
-  fontsBuffers: Uint8Array[] // A list of raw font files to load.
-} & FontOptions
-
-export type SystemFontsOptions = {
-  loadSystemFonts?: boolean // Default: true. if set to false, it will be faster.
-  fontFiles?: string[] // A list of local font file paths to load.
-  fontDirs?: string[] // A list of local font directories to load.
-} & FontOptions
-
```
Comment (yisibl) [inline index.d.ts]: There is no `fontsBuffers` option in `index.d.ts`, which is currently specific to Wasm.

File: index.d.ts [hunk 3]
```diff
@@ -29,27 +40,6 @@ export type ResvgRenderOptions = {
...
-  serifFamily?: string
-  sansSerifFamily?: string
-  cursiveFamily?: string
-  fantasyFamily?: string
-  monospaceFamily?: string
-}
-
-export type CustomFontsOptions = {
-  fontsBuffers: Uint8Array[] // A list of raw font files to load.
-} & FontOptions
```
Comment (yisibl) [inline index.d.ts]: There is no fontsBuffers option in index.d.ts, which is currently specific to Wasm.

**Comment by lasseschou** [comment]
Great work. @yisibl can you please release a new version so we can start using fonts? 👍

**Comment by yisibl** [comment]
@lasseschou Please wait for me to update the documentation and playground.

**Comment by yisibl** [comment]
This PR has been released to v2.5.0, please note that the option name has been changed from `fontsBuffers` to `fontBuffers`.

**Pull request merged**